storagenode/storagenodedb/pieceinfo.go

```diff
@@ -47,6 +47,31 @@ func (db *pieceinfo) Add(ctx context.Context, info *pieces.Info) (err error) {
 	return ErrInfo.Wrap(err)
 }
 
+// GetPiecesID gets piece information by satellite id.
+func (db *pieceinfo) GetPiecesID(ctx context.Context, satelliteID storj.NodeID, createdBefore time.Time, limit, offset int) (pieceIDs []storj.PieceID, err error) {
+	defer mon.Task()(&ctx)(&err)
+
+	rows, err := db.db.QueryContext(ctx, db.Rebind(`
+		SELECT piece_id
+		FROM pieceinfo
+		WHERE satellite_id = ? AND piece_creation < ?
+		LIMIT ? OFFSET ?
+	`), satelliteID, createdBefore, limit, offset)
```

Comment:
We should subtract an extra hour from `createdBefore` so we leave some room for clock difference between the satellite and the storage node. I believe we had this recommendation in the design doc.

Reply:
Fadila said this in slack but in case anyone is here wondering, it looks like subtracting one hour isn't necessary because the creation time is fixed by the satellite in order limits